custom/icds_reports/management/commands/phone_info_grouped_by_state_district.py

```diff
@@ -0,0 +1,150 @@
+    END) "phone_not_available"
+ """
+
+COMMON_QUERY_END = """ and person_cases.migration_status=0 and person_cases.closed_on is null and person_cases.registered_status=1
+group by
+person_cases.state_id, person_cases.district_id"""
+
+CONDITION_PREGNANT_WOMAN = """ FROM "ucr_icds-cas_static-person_cases_v3_2ae0879a" person_cases
+where person_cases.is_pregnant=1 """
+
+CONDITION_LACTATING_MOTHER = """ FROM "ucr_icds-cas_static-ccs_record_cases_cedcca39" ccs_record
+LEFT OUTER JOIN "ucr_icds-cas_static-person_cases_v3_2ae0879a" person_cases
+ON
+    ccs_record.person_case_id = person_cases.doc_id
+    AND
+    ccs_record.supervisor_id = person_cases.supervisor_id
+where ccs_record.add is not null AND now() - ccs_record.add  < '182 days' ::interval """
+
+CONDITION_CHILDREN = """ FROM "ucr_icds-cas_static-child_health_cases_a46c129f" child_health
+LEFT OUTER JOIN "ucr_icds-cas_static-person_cases_v3_2ae0879a" person_cases
+ON
+    child_health.mother_id = person_cases.doc_id
+    AND
+    child_health.state_id = person_cases.state_id
+    AND
+    child_health.supervisor_id = person_cases.supervisor_id
+where child_health.dob is NOT null AND now() - child_health.dob <= '2191.5 days' ::interval """
+
+CONDITION_WOMEN_LESS_THAN_45 = """ FROM "ucr_icds-cas_static-person_cases_v3_2ae0879a" person_cases
+where  person_cases.sex = 'F' AND now() - person_cases.dob >= '4017.75 days' ::interval and now() - person_cases.dob <= '17897.25 days' :: interval """
+
+CONDITION_ALL_WOMEN = """ FROM "ucr_icds-cas_static-person_cases_v3_2ae0879a" person_cases
+where  person_cases.sex = 'F' AND now() - person_cases.dob >= '4017.75 days' ::interval """
+
+BENEFICIARY_TYPES = ["PW", "LM", "Children", "Women Less than 45 yrs", "All Women"]
+
+CSV_FILENAME = "phone_number_info_grouped_by_districts_{datetime}.csv".format(
+    datetime=datetime.utcnow()
+)
+
+
+class Command(BaseCommand):
+    def prepare_query(self, condition):
+        return COMMON_QUERY_START + condition + COMMON_QUERY_END
+
+    def get_loc_name(self, loc):
+        if self.location_to_name.get(loc):
+            return self.location_to_name[loc]
+        location = SQLLocation.by_location_id(loc)
+        self.location_to_name[loc] = location.name
+        return location.name
+
+    def get_variable_headers(self):
+        return ['Count', 'Valid Phone Numbers', 'Invalid Phone Numbers', 'No Phone Numbers']
+
+    def get_fixed_header_names(self):
+        return ['State', 'District']
+
+    def get_location_key(self, state_id, district_id):
+        return state_id + '_' + district_id
+
+    def get_placeholders_for_empty_rows(self, count):
+        placeholders = ()
+        for i in range(count):
+            placeholders += 0,
+        return placeholders
+
+    def append_data_to_result(self, row, count):
+        key = self.get_location_key(row[0], row[1])
+        state = self.get_loc_name(row[0])
+        district = self.get_loc_name(row[1])
+        expected_entries = (count * 4) + 2
```

Comment:
2 rows are reserverd for state and district and each category has 4 entries.
If previous query does not have any entry for the given category we are filling it with 0s so that column order is maintained.